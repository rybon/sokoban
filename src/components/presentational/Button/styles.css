@import 'styles/variables';

.button {
  background-color: #666;
  color: white;
  text-align: center;
  padding: 5px;
  border-radius: 5px;
}

.selected {
  background-color: var(--buttonColor);
}
