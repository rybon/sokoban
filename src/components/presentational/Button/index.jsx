import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './styles.css'

export default class Button extends PureComponent {
  static propTypes = {
    selected: PropTypes.bool,
    className: PropTypes.string,
    onClick: PropTypes.func,
    children: PropTypes.string
  }

  static defaultProps = {
    selected: false,
    className: '',
    onClick: () => ({}),
    children: ''
  }

  render() {
    const { selected, className, onClick, children } = this.props

    return (
      <div
        className={classNames(
          styles.button,
          selected && styles.selected,
          className
        )}
        role="presentation"
        onClick={onClick}
      >
        {children}
      </div>
    )
  }
}
