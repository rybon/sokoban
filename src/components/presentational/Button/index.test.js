import React from 'react'
import sinon from 'sinon'
import { mount } from 'enzyme'
import Button from './index'

describe('<Button />', () => {
  const onClick = sinon.spy()

  const button = mount(<Button selected onClick={onClick} />)

  it('renders correctly', () => {
    expect(button).toMatchSnapshot()
  })

  it('responds to click events', () => {
    button.find('div').simulate('click')

    expect(onClick.called).toBe(true)
  })
})
