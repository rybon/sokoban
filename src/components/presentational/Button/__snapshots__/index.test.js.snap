// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<Button
  className=""
  onClick={[Function]}
  selected={true}
>
  <div
    className="button selected"
    onClick={[Function]}
    role="presentation"
  />
</Button>
`;
